lib/perl/Genome/Search/Queue.pm

```diff
@@ -79,4 +82,70 @@ sub default_priority {
     return $property->{default_value};
 }
 
+sub create_dedup_iterator {
+    return $_[0]->create_iterator(
+        -group_by => [qw(subject_class subject_id)],
+    );
+}
+
+sub dedup_set {
+    my ($class, $set) = Params::Validate::validate_pos(@_,
+        { type => SCALAR },
+        { isa => 'Genome::Search::Queue::Set',
+          callbacks => {
+              'specifies subject_id' => sub { shift->rule->value_for('subject_id') },
+              'specifies subject_class' => sub { shift->rule->value_for('subject_class') },
+          },
+        },
+    );
+
+    my $m_iter = $set->member_iterator;
+    my $q = $m_iter->next;
+    my $duplicate_count = 0;
+    while (my $q = $m_iter->next) {
```

Comment:
It strikes me as odd that both the one to keep and the ones to delete are kept in `my $q`.